src/gather/api.py

```diff
@@ -49,6 +50,23 @@ class Collector(object):
     and *collect*-ing them when they need to be used.
     """
 
+    @staticmethod
+    def one_of(_registry, _effective_name, objct):
```

Comment:
Docstring?  Also, a `staticmethod`?  Why not a module-level function?  That would remove the need to grab `__func__` below.